### PR TITLE
feat: add anchor

### PR DESCRIPTION
**Pull request by duidae**
[ticket](https://app.asana.com/0/1205517559975955/1206167732498118/f)

1. 在draft-renderer/draft-editor新增錨點anchor decorator(上次的索引改名為toc-anchor)
2. 修改draft-renderer的link decorator, 由url判斷內外部連結render不同的link
3. 前端新增若網址包含anchor則滑動到對應錨點的邏輯

### CODE REVIEW

File: packages/draft-renderer/src/draft-renderer.tsx [hunk 1]
```diff
@@ -26,43 +27,62 @@ type DraftRendererProps = {
...
-      <Editor
-        editorState={editorState}
-        blockRenderMap={blockRenderMaps.content}
-        blockRendererFn={blockRendererFn}
-        customStyleFn={customStyleFn}
-        readOnly
-        // eslint-disable-next-line @typescript-eslint/no-empty-function
-        onChange={() => {}}
-      />
+      <TopOffsetContext.Provider value={offsetTop}>
```
Comment (nickhsine) [inline packages/draft-renderer/src/draft-renderer.tsx]: Question:
要不要把 `offsetTop` 放進 `ThemeProvider` 裡面？

File: packages/frontend/src/app/(sticky-header)/article/[slug]/hero-image.tsx [hunk 2]
```diff
@@ -7,19 +7,21 @@ type HeroImageProp = {
...
-          src={image?.resized?.medium}
-        />
-      </div>
-      <figcaption>{caption}</figcaption>
-    </figure>
+    image && (
+      <figure className="hero-image">
+        <div className="image-container">
+          <img
+            srcSet={`${image.resized?.small} 800w, ${image.resized?.medium} 1200w, ${image.resized?.large} 2000w"`}
```
Comment (nickhsine) [inline packages/frontend/src/app/(sticky-header)/article/[slug]/hero-image.tsx]: issue:
目前的寫法會產生解析度不足的問題，請參考：https://app.asana.com/0/1205517559975955/1206685336539536/f

File: packages/draft-renderer/src/draft-renderer.tsx (hunk 1 above)
Comment (duidae) [reply packages/draft-renderer/src/draft-renderer.tsx]: moved to ThemeProvider  in https://github.com/kids-reporter/kids-reporter-monorepo/pull/493/commits/d8b48a9d69add8d6f8de4e6d6953657e1dce120c

File: packages/frontend/src/app/(sticky-header)/article/[slug]/hero-image.tsx (hunk 2 above)
Comment (duidae) [reply packages/frontend/src/app/(sticky-header)/article/[slug]/hero-image.tsx]: fixed in https://github.com/kids-reporter/kids-reporter-monorepo/pull/493/commits/8507250d1d2f6c511b1cab27bba7b4a3b371bcf6

**Comment by nickhsine** [comment]
LGTM 👍 👍 👍

File: packages/draft-renderer/src/entity-decorators/toc-anchor.tsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+const TOCAnchor = (props: {
+  decoratedText: string
+  contentState: ContentState
+  entityKey: string
+  children: React.ReactNode
+}) => {
+  const { children, contentState, entityKey } = props
+  const key = contentState?.getEntity(entityKey)?.getData()?.anchorKey
+  return <span id={`anchor-${key}`}>{children}</span>
```
Comment (nickhsine) [inline packages/draft-renderer/src/entity-decorators/toc-anchor.tsx]: nitpick:
anchor 和 tocAnchor 好像有點容易造成混肴？
像是此處應該是 id={\`anchor-${key}\`} 還是 id={\`toc-anchor-${key}\`} 呢？

Comment (duidae) [reply packages/draft-renderer/src/entity-decorators/toc-anchor.tsx]: refactored in https://github.com/kids-reporter/kids-reporter-monorepo/pull/493/commits/198a1b30b45e50ddec83f9a5415d736effd8dcfd

**Comment by nickhsine** [comment]
LGTM 👍

**Pull request merged**